src/main/java/net/dv8tion/jda/bot/sharding/DefaultShardManager.java

```diff
@@ -770,4 +764,28 @@ protected ScheduledExecutorService createExecutor(ThreadFactory threadFactory)
 
         return Executors.newSingleThreadScheduledExecutor(factory);
     }
+
+    protected static <E extends ExecutorService> ExecutorPaid<E> getExecutor(ThreadPoolProvider<? extends E> provider, int shardId)
```

Comment:
`ExecutorPair` instead of `ExecutorPaid` perhaps?

Reply:
:man_facepalming: